BUILD.md

```diff
@@ -10,33 +10,42 @@ for C++.  Applications heavily leveraging Kokkos are strongly encouraged to use
 You can either use Kokkos as an installed package (encouraged) or use Kokkos in-tree in your project.
 Modern CMake is exceedingly simple at a high-level (with the devil in the details).
 Once Kokkos is installed In your `CMakeLists.txt` simply use:
-````
+````cmake
 find_package(Kokkos REQUIRED)
 ````
 Then for every executable or library in your project:
-````
+````cmake
 target_link_libraries(myTarget Kokkos::kokkos)
 ````
 That's it! There is no checking Kokkos preprocessor, compiler, or linker flags.
 Kokkos propagates all the necessary flags to your project.
 This means not only is linking to Kokkos easy, but Kokkos itself can actually configure compiler and linker flags for *your*
-project. If building in-tree, there is no `find_package` and you link with `target_link_libraries(kokkos)`.
+project.
+When configuring your project just set:
+````bash
+> cmake -DKokkos_ROOT=${kokkos_install_prefix}
+````
+Note: You may need the following if using some versions of CMake (e.g. 3.12):
+````cmake
+cmake_policy(SET CMP0074 NEW)
+````
+If building in-tree, there is no `find_package`. You can use `add_subdirectory(kokkos)` with the Kokkos source and again just link with `target_link_libraries(Kokkos::kokkos)`.
```

Comment:
I would mention `examples/cmake_build_installed` and `examples/cmake_build_in_tree` here as examples.

Reply:
Added. I had to clean up these examples as well (redundant with another PR, but we should fix the comments in these examples ASAP).